tests/test_tomography_gate_set_tomography.py

```diff
@@ -0,0 +1,392 @@
+            [gates.M(0, basis=gates.X), gates.M(1)],
+            [gates.M(0, basis=gates.X), gates.M(1, basis=gates.X)],
+            [gates.M(0, basis=gates.X), gates.M(1, basis=gates.Y)],
+            [gates.M(0, basis=gates.X), gates.M(1)],
+            [gates.M(0, basis=gates.Y), gates.M(1)],
+            [gates.M(0, basis=gates.Y), gates.M(1, basis=gates.X)],
+            [gates.M(0, basis=gates.Y), gates.M(1, basis=gates.Y)],
+            [gates.M(0, basis=gates.Y), gates.M(1)],
+            [gates.M(0), gates.M(1)],
+            [gates.M(0), gates.M(1, basis=gates.X)],
+            [gates.M(0), gates.M(1, basis=gates.Y)],
+            [gates.M(0), gates.M(1)],
+        ],
+    }
+    errors = {(0, 3): ValueError, (17, 1): IndexError}
+    if (j, nqubits) in [(0, 3), (17, 1)]:
+        with pytest.raises(errors[(j, nqubits)]):
+            prepared_gates = _measurement_basis(j, nqubits)
+    else:
+        prepared_gates = _measurement_basis(j, nqubits)
+        for groundtruth, gate in zip(correct_gates[nqubits][j], prepared_gates):
+            _compare_gates(groundtruth, gate)
+            for g1, g2 in zip(groundtruth.basis, gate.basis):
+                _compare_gates(g1, g2)
+
+
+@pytest.mark.parametrize(
+    "j,nqubits",
+    INDEX_NQUBITS,
+)
+def test__get_observable(j, nqubits):
+    correct_observables = {
+        1: [
+            (S(1),),
+            (qibo.symbols.Z(0),),
+            (qibo.symbols.Z(0),),
+            (qibo.symbols.Z(0),),
+        ],
+        2: [
+            (S(1), S(1)),
+            (S(1), qibo.symbols.Z(1)),
+            (S(1), qibo.symbols.Z(1)),
+            (S(1), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), S(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), S(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), S(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+            (qibo.symbols.Z(0), qibo.symbols.Z(1)),
+        ],
+    }
+    correct_observables[1] = [
+        SymbolicHamiltonian(h[0]).form for h in correct_observables[1]
+    ]
+    correct_observables[2] = [
+        SymbolicHamiltonian(reduce(lambda x, y: x * y, h)).form
+        for h in correct_observables[2]
+    ]
+    errors = {(0, 3): ValueError, (17, 1): IndexError}
+    if (j, nqubits) in [(0, 3), (17, 1)]:
+        with pytest.raises(errors[(j, nqubits)]):
+            prepared_observable = _get_observable(j, nqubits)
+    else:
+        prepared_observable = _get_observable(j, nqubits).form
+        groundtruth = correct_observables[nqubits][j]
+        assert groundtruth == prepared_observable
+
+
+def test_expectation_value(backend):
```

Comment:
I am not sure how useful is this test, we should either add a proper complete test for `_expectation_value` or just add to the `test_GST` test this error case.

Reply:
Are there >=3 qubit gates that we can use for the `_expectation_value` test?

Reply:
Ah ok, because you are just testing for the error to be raised with 3 qubits, but then give a proper name to the test, something like `test__expectation_value_nqubits_error`, otherwise one might think that you are testing the `_expectation_value` function which you are not.